dbt_dry_run/node_runner/model_runner.py

```diff
@@ -77,6 +84,13 @@ def _modify_sql(self, node: Node, sql_statement: str) -> str:
         if node.config.materialized == "view":
             sql_statement = f"CREATE OR REPLACE VIEW `{node.database}`.`{node.db_schema}`.`{node.alias}` AS (\n{sql_statement}\n)"
 
+        if node.config.materialized == "incremental" and node.config.partition_by:
```

Comment:
To be more correct the materialisation only includes this variable if `"_dbt_max_partition" in node.compiled_sql` so another condition here in this if statement would mirror dbt's behaviour more closely